Helper/Data.php

```diff
@@ -411,12 +425,13 @@ public function callApi($endpoint, $data = [], $method = 'GET')
         $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
         $curl = $objectManager->create('\Magento\Framework\HTTP\Client\Curl');
         try {
-            $data = http_build_query($data);
-            if ($method = 'GET') {
+            if ($method === 'GET') {
+                $data = http_build_query($data);
                 $apiUrl = $apiUrl . "?" . $data;
+                $curl->setCredentials($username, $password);
                 $curl->get($apiUrl);
             }
-            if ($method = 'POST') {
+            if ($method === 'POST') {
```

Comment:
This isn't an issue, but it is a good practice to write if-conditions that cannot be true at the same time in `elseif` blocks.

Reply:
Usin `elseif ` block
https://github.com/sendsmaily/smaily-magento-extension/pull/12/commits/0cfb9706c277c4d2887914376049a2106ad5dbe3